src/components/ui/chart.tsx

```
@@ -0,0 +1,353 @@
+// Format: { THEME_NAME: CSS_SELECTOR }
+const THEMES = { light: "", dark: ".dark" } as const
+
+export type ChartConfig = {
+  [k in string]: {
+    label?: React.ReactNode
+    icon?: React.ComponentType
+  } & (
+    | { color?: string; theme?: never }
+    | { color?: never; theme: Record<keyof typeof THEMES, string> }
+  )
+}
```

ESLint

[error] A record is preferred over an index signature.

(@typescript-eslint/consistent-indexed-object-style)